core/system/src/mpu/vmpu_armv7m.c

```diff
@@ -329,52 +349,68 @@ void vmpu_switch(uint8_t src_box, uint8_t dst_box)
         HALT_ERROR(SANITY_CHECK_FAILED, "ACL add: The destination box ID is out of range (%u).\r\n", dst_box);
     }
 
-    /* update target box first to make target stack available */
-    mpu_slot = ARMv7M_MPU_RESERVED_REGIONS;
+    box = &g_mpu_box[dst_box];
+
+    /* Update target box first to make target stack available. */
+    g_mpu_slot = ARMv7M_MPU_REGIONS_STATIC;
+    g_mpu_slot_wrap_around = ARMv7M_MPU_REGIONS_STATIC;
+    region = box->region;
+    dst_count = box->count;
+
+    /* Only write stack and context ACL for secure boxes. */
     if (dst_box)
     {
-        /* handle target box next */
-        box = &g_mpu_box[dst_box];
-        region = box->region;
+        assert(dst_count);
```

Comment:
@niklas-arm moving the assert outside of the if-block (above) - as it's used also in other blocks.